core/services/ocr2/plugins/functions/integration_tests/functions_integration_test.go

```diff
@@ -89,8 +89,114 @@ func TestIntegration_Functions_MultipleRequests_Success(t *testing.T) {
 				answer, err := clientContracts[ic].Contract.LastResponse(nil)
 				require.NoError(t, err)
 				return answer
-			}, 1*time.Minute, 1*time.Second).Should(gomega.Equal(utils.GetExpectedResponse(requestSources[ic])))
+			}, 3*time.Minute, 1*time.Second).Should(gomega.Equal(utils.GetExpectedResponse(requestSources[ic])))
 		}()
 	}
 	wg.Wait()
+	fmt.Println("Done with basic tests")
+}
+
+func TestIntegration_Functions_MultipleRequests_SuccessWithThresholdDecryption(t *testing.T) {
```

Comment:
Maybe move the test to a separate file? It's already cluttered. Put all the encryption-related constants at the top with lengthy explanations.